### DISCUSSION TITLE
MySQL panic: range end index 4 out of range for slice of length 2

### DISCUSSION THREAD

**Discussion by alexanderjophus** [diesel-rs/diesel#3975] [Q&A] [answered]
Hey, firstly thanks for an amazing crate, it's working wonders for me.

I have a MySQL schema with a table, and a column defined as
```mysql
    `HP lvl 1` BIGINT,
```

This then generates a schema including
```rs
        HP_lvl_1 -> Nullable<Bigint>,
```

I also have a models.rs, (which is a copy almost verbatim of the schema table)
```rs
    pub HP_lvl_1: Option<i64>,
```

I'm using a simple `.load()` function to load all the data (just testing things atm)

I receive the following panic
```
"panic.payload":"range end index 4 out of range for slice of length 2",
"panic.location":"...cargo/registry/src/index.crates.io-6f17d22bba15001f/diesel-2.1.5/src/mysql/value.rs:69:76"
```

Clicking into the schema Bigint, I see the following macro declared
```rs
#[diesel(mysql_type(name = "LongLong"))]
```

However, the panic is occurring at this point
```rs
            MysqlType::UnsignedLong | MysqlType::Long => {
                NumericRepresentation::Medium(i32::from_ne_bytes((&self.raw[..4]).try_into()?))
            }
```

It seems to me, that there's a disconnect where the schema and model are `LongLong` and i64 respectively, but the parsing in the crate is hitting `Long` and parsing into i32.

**Comment by weiznich** [comment]
Thanks for opening this discussion. This might be a bug in diesel. Could you provide some code + schema to reproduce the issue? Also information about your exact MySQL and mysqlclient version might help to track the issue down.

**Reply by alexanderjophus** [reply]
Thanks for the speedy response!

https://github.com/alexanderjophus/cyber-sleuth

This is the repo I'm working on. I'm hacking together a few things, but I can probably make a minimum reproducible repo. I'll try to guide you through the repo I'm working on first as I'm not that far into it.

This is the query that's failing: https://github.com/alexanderjophus/cyber-sleuth/blob/main/app/src/routes/mod.rs#L43-L45

This is the models.rs: https://github.com/alexanderjophus/cyber-sleuth/blob/main/app/src/models.rs

This is the schema.rs that was autogenerated: https://github.com/alexanderjophus/cyber-sleuth/blob/main/app/src/schema.rs

Not entirely too comfortable with mysql, so may need help getting version info:

```sh
❯ mysql --version
mysql  Ver 8.3.0 for macos12.6 on x86_64 (Homebrew)
```

Hope that helps!

**Reply by weiznich** [reply, marked as answer]
That sounds related to https://github.com/diesel-rs/diesel/issues/3929. It seems like there are a few backward incompatible changes in the MySQL 8.3 release.

**Reply by alexanderjophus** [reply]
Looks like that was it, thank you!

**Reply by weiznich** [reply]
Thanks for confirming this. Good to know that downgrading helps. Nevertheless we should work on fixing the actual problem. I would like to use #3929 to track that.
